metadata-ingestion/src/datahub/sdk/datajob.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+import warnings
+from datetime import datetime
+from typing import Dict, Optional, Type
+
+from typing_extensions import Self
+
+from datahub.cli.cli_utils import first_non_null
+from datahub.errors import IngestionAttributionWarning
+from datahub.metadata.schema_classes import (
+    AspectBag,
+    AzkabanJobTypeClass,
+    DataJobInfoClass,
+    EditableDataJobPropertiesClass,
+)
+from datahub.metadata.urns import DataJobUrn, Urn
+from datahub.sdk._attribution import is_ingestion_attribution
+from datahub.sdk._shared import (
+    DataflowUrnOrStr,
+    DomainInputType,
+    HasContainer,
+    HasDomain,
+    HasInstitutionalMemory,
+    HasOwnership,
+    HasPlatformInstance,
+    HasSubtype,
+    HasTags,
+    HasTerms,
+    LinksInputType,
+    OwnersInputType,
+    TagsInputType,
+    TermsInputType,
+    make_time_stamp,
+    parse_time_stamp,
+)
+from datahub.sdk.entity import Entity, ExtraAspectsType
+
+
+class DataJob(
+    HasPlatformInstance,
+    HasSubtype,
+    HasContainer,
+    HasOwnership,
+    HasInstitutionalMemory,
+    HasTags,
+    HasTerms,
+    HasDomain,
+    Entity,
+):
+    """Represents a data job in DataHub.
+    A data job is an executable unit of a data pipeline, such as an Airflow task or a Spark job.
+    """
+
+    __slots__ = ()
+
+    @classmethod
+    def get_urn_type(cls) -> Type[DataJobUrn]:
+        """Get the URN type for data jobs."""
+        return DataJobUrn
+
+    def __init__(
+        self,
+        *,
+        # Identity
+        id: str,
+        flow_urn: DataflowUrnOrStr,
```

Comment:
imo this feels pretty clunky - ideally we should have simpler ways of creating a dataflow and a bunch of child datajobs

I don't think the existing classes are simple enough